lc-hax/Scripts/Modules/Enemy Controls/Enemies/BaboonController.cs

```diff
@@ -0,0 +1,67 @@
+using Unity.Netcode;
+using UnityEngine;
+
+public static class BaboonController {
+
+
+    public static void UseSecondarySkill(this BaboonBirdAI instance) {
+        if (instance is null) return;
+        if (instance.heldScrap == null) {
+            instance.GrabNearbyItem();
+        }
+        else {
+            instance.DropCurrentItem();
+        }
+    }
+
+    public static NetworkObject? FindNearbyItem(this BaboonBirdAI instance, float range = 1.5f) {
+        if (instance is null) return null;
+        Collider[] Search = Physics.OverlapSphere(instance.gameObject.transform.position, range);
+        for (int i = 0; i < Search.Length; i++) {
+            if (Search[i].TryGetComponent(out GrabbableObject item)) {
+                if (instance.CanGrabScrap(item)) {
+                    if (item.TryGetComponent(out NetworkObject network)) {
+                        float dist = Vector3.Distance(item.transform.position, instance.transform.position);
+                        if (dist < float.PositiveInfinity) {
+                            return network;
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Fantastic work. Could you reduce the nesting here and other similar places? I believe I saw some names that were too generic as well.

Reply:
i have been on this for 5 hours, i tried to reduce the nesting as much i could, i dont think is possible unless you want to give a try

Reply:
Take a break. Also, you can use the `continue` statement.

Reply:
> Take a break. Also, you can use the `continue` statement.

why i need to use the continue statement when i return the object from that method? im confused lol

Reply:
``cs
if (!Search[i].TryGetComponent(out GrabbableObject item)) continue;
 ```

This is known as an if-guard.